session-helper/flatpak-session-helper.c

```diff
@@ -425,24 +447,88 @@ file_changed (GFileMonitor     *monitor,
               GFile            *file,
               GFile            *other_file,
               GFileMonitorEvent event_type,
-              char             *source)
+              MonitorData      *data)
 {
-  if (event_type == G_FILE_MONITOR_EVENT_CHANGES_DONE_HINT ||
-      event_type == G_FILE_MONITOR_EVENT_CREATED)
-    copy_file (source, monitor_dir);
+  if (event_type != G_FILE_MONITOR_EVENT_CHANGES_DONE_HINT)
```

Comment:
Does `GFileMonitor` guarantee that `EVENT_CHANGES_DONE_HINT` will always be emitted? If not, I’d be tempted to listen to `EVENT_CREATED` too, like the code used to, to hedge against regressions there.

Reply:
Both Linux backends I can see (inotify and polling) defensively emit DONE_HINT (in the CREATED or every case respectively), and are wrapped in a GFileMonitorSource which also has a timer mechanism to ensure that the DONE_HINT is also given for every file a certain time after the last modification. So I think this is a pretty safe bet.

Reply:
Thanks for checking 👍